node/impl/full/eth.go

```diff
@@ -1455,77 +1455,23 @@ func lookupEthAddress(ctx context.Context, addr address.Address, sa StateAPI) (e
 }
 
 func newEthTxFromFilecoinMessage(ctx context.Context, smsg *types.SignedMessage, sa StateAPI) (ethtypes.EthTx, error) {
-	fromEthAddr, err := lookupEthAddress(ctx, smsg.Message.From, sa)
-	if err != nil {
-		return ethtypes.EthTx{}, err
-	}
-
-	toEthAddr, err := lookupEthAddress(ctx, smsg.Message.To, sa)
+	tx, err := ethtypes.EthTxFromSignedMessage(smsg)
 	if err != nil {
-		return ethtypes.EthTx{}, err
+		return ethtypes.EthTx{}, xerrors.Errorf("failed to convert from signed message: %w", err)
 	}
 
-	toAddr := &toEthAddr
-	input := smsg.Message.Params
-	// Check to see if we need to decode as contract deployment.
-	// We don't need to resolve the to address, because there's only one form (an ID).
-	if smsg.Message.To == builtintypes.EthereumAddressManagerActorAddr {
-		switch smsg.Message.Method {
-		case builtintypes.MethodsEAM.Create:
-			toAddr = nil
-			var params eam.CreateParams
-			err = params.UnmarshalCBOR(bytes.NewReader(smsg.Message.Params))
-			input = params.Initcode
-		case builtintypes.MethodsEAM.Create2:
-			toAddr = nil
-			var params eam.Create2Params
-			err = params.UnmarshalCBOR(bytes.NewReader(smsg.Message.Params))
-			input = params.Initcode
-		case builtintypes.MethodsEAM.CreateExternal:
-			toAddr = nil
-			var params abi.CborBytes
-			err = params.UnmarshalCBOR(bytes.NewReader(smsg.Message.Params))
-			input = []byte(params)
-		}
-		if err != nil {
-			return ethtypes.EthTx{}, err
-		}
-	}
-	// Otherwise, try to decode as a cbor byte array.
-	// TODO: Actually check if this is an ethereum call. This code will work for demo purposes, but is not correct.
-	if toAddr != nil {
-		if decodedParams, err := cbg.ReadByteArray(bytes.NewReader(smsg.Message.Params), uint64(len(smsg.Message.Params))); err == nil {
-			input = decodedParams
-		}
-	}
-
-	r, s, v, err := ethtypes.RecoverSignature(smsg.Signature)
+	fromEthAddr, err := lookupEthAddress(ctx, smsg.Message.From, sa)
 	if err != nil {
-		// we don't want to return error if the message is not an Eth tx
-		r, s, v = ethtypes.EthBigIntZero, ethtypes.EthBigIntZero, ethtypes.EthBigIntZero
+		return ethtypes.EthTx{}, xerrors.Errorf("failed to lookup from ethaddress: %w", err)
 	}
 
-	hash, err := ethtypes.EthHashFromCid(smsg.Cid())
+	toEthAddr, err := lookupEthAddress(ctx, smsg.Message.To, sa)
```

Comment:
If this is a valid eth transaction, `to` should either be an ID address, or an f410 address. We shouldn't need to resolve this if we're going to be limiting support to _real_ eth transactions.